ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -169,7 +260,7 @@ private void disconnect() throws XAException {
             xaConnection.close();
         } catch (JMSException e) {
             LOGGER.warn("Failed to close connection", e);
-            throw new XAException(e.getMessage());
+            throw new XAException(XAException.XAER_PROTO);
```

Comment:
I agree:
    case XAException.XAER_RMFAIL:
	 _committed = true;  // will cause log to be rewritten